code/obj/submachine/weapon_vendor.dm

```diff
@@ -21,15 +21,78 @@
 	density = 1
 	opacity = 0
 	anchored = 1
+	flags = TGUI_INTERACTIVE
 
 	var/sound_token = 'sound/machines/capsulebuy.ogg'
 	var/sound_buy = 'sound/machines/spend.ogg'
-	var/temp = null
 	var/list/credits = list("Sidearm" = 0, "Loadout" = 0, "Utility" = 0, "Assistant" = 0)
 	var/list/datum/materiel_stock = list()
 	var/token_accepted = /obj/item/requisition_token
 	var/log_purchase = FALSE
 
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "WeaponVendor", src.name)
+			ui.open()
+
+	ui_static_data(mob/user)
+		. = list("stock" = list())
+
+		for (var/datum/materiel/M in materiel_stock)
+			.["stock"] += list(list(
+				"ref" = "\ref[M]",
+				"name" = M.name,
+				"description" = M.description,
+				"cost" = M.cost,
+				"category" = M.category,
+			))
+
+	ui_data(mob/user)
+		. = list(
+			"credits" = src.credits,
+		)
+
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		. = ..()
+		if (.)
+			return
+
+		switch(action)
+			if ("redeem")
+				var/datum/materiel/sidearm/S = locate(params["ref"]) in materiel_stock
+				if(istype(S))
+					if(src.credits["Sidearm"] >= S.cost)
+						src.credits["Sidearm"] -= S.cost
+						var/atom/A = new S.path(src.loc)
+						playsound(src.loc, sound_buy, 80, 1)
+						src.vended(A)
+						return TRUE
```

Comment:
the code duplication is annoying, but not yet at the point where i would demand a separation of this out

Reply:
Fixed it anyways